kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -67,32 +83,106 @@ public SchemaManager(
   /**
    * Create a new table in BigQuery.
    * @param table The BigQuery table to create.
-   * @param topic The Kafka topic used to determine the schema.
+   * @param records The sink records used to determine the schema.
    */
-  public void createTable(TableId table, String topic) {
-    Schema kafkaValueSchema = schemaRetriever.retrieveSchema(table, topic, KafkaSchemaRecordType.VALUE);
-    Schema kafkaKeySchema = kafkaKeyFieldName.isPresent() ? schemaRetriever.retrieveSchema(table, topic, KafkaSchemaRecordType.KEY) : null;
-    bigQuery.create(constructTableInfo(table, kafkaKeySchema, kafkaValueSchema));
+  public void createTable(TableId table, Set<SinkRecord> records) {
+      TableInfo tableInfo = getTableInfo(table, records);
+      bigQuery.create(tableInfo);
   }
 
   /**
    * Update an existing table in BigQuery.
    * @param table The BigQuery table to update.
-   * @param topic The Kafka topic used to determine the schema.
+   * @param records The sink records used to update the schema.
    */
-  public void updateSchema(TableId table, String topic) {
-    Schema kafkaValueSchema = schemaRetriever.retrieveSchema(table, topic, KafkaSchemaRecordType.VALUE);
-    Schema kafkaKeySchema = kafkaKeyFieldName.isPresent() ? schemaRetriever.retrieveSchema(table, topic, KafkaSchemaRecordType.KEY) : null;
-    TableInfo tableInfo = constructTableInfo(table, kafkaKeySchema, kafkaValueSchema);
+  public void updateSchema(TableId table, Set<SinkRecord> records) {
+    TableInfo tableInfo = getTableInfo(table, records);
     logger.info("Attempting to update table `{}` with schema {}",
-        table, tableInfo.getDefinition().getSchema());
+            table, tableInfo.getDefinition().getSchema());
     bigQuery.update(tableInfo);
   }
 
-  // package private for testing.
-  TableInfo constructTableInfo(TableId table, Schema kafkaKeySchema, Schema kafkaValueSchema) {
-    com.google.cloud.bigquery.Schema bigQuerySchema = getBigQuerySchema(kafkaKeySchema, kafkaValueSchema);
+  private TableInfo getTableInfo(TableId table, Set<SinkRecord> records) {
```

Comment:
Could you add some Javadocs on these new private methods? I think they're sufficiently complicated to warrant some docs.

Reply:
Sure. Will update.